drivers: eeprom: at2x: use new DT_INST macros

henrikbrixandersen opened the pull request:
Convert older DT_INST_ macro use the new include/devicetree.h DT_INST macro APIs.

Signed-off-by: Henrik Brix Andersen <hebad@vestas.com>

henrikbrixandersen committed:
drivers: eeprom: at2x: use new DT_INST macros

Convert older DT_INST_ macro use the new include/devicetree.h
DT_INST macro APIs.

Signed-off-by: Henrik Brix Andersen <hebad@vestas.com>

gmarull commented on drivers/eeprom/eeprom_at2x.c:
```diff
@@ -550,138 +550,62 @@ static const struct eeprom_driver_api eeprom_at2x_api = {
-EEPROM_AT2X_DEVICE(24, 1);
-#endif
+#define DT_INST_AT2X_FOREACH(t, inst_expr) \
+	UTIL_LISTIFY(DT_NUM_INST(atmel_at##t), DT_CALL_WITH_ARG, inst_expr)
```
Should likely wait for https://github.com/zephyrproject-rtos/zephyr/pull/23416

henrikbrixandersen replied:
No, it can’t. This driver supports multiple, different compatibles.

henrikbrixandersen committed:
tests: drivers: eeprom: use new DT_ALIAS macros

Convert older DT_ALIAS_ macro use the new include/devicetree.h
DT_ALIAS macro APIs.

Signed-off-by: Henrik Brix Andersen <hebad@vestas.com>

henrikbrixandersen committed:
tests: drivers: eeprom: do not attempt to build on all boards

Do not attempt to build on all boards. Instead, do a build-only pass
of the generic test case on all boards with EEPROM support.

Signed-off-by: Henrik Brix Andersen <hebad@vestas.com>
